plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1012,6 +1012,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                  transaction_metadata_ptr trx_meta;
                  try {
                     trx_meta = transaction_metadata::recover_keys(trx, chain.get_chain_id(), time_limit, trx_type,
+                                                                  chain.check_canonical(),
```

Comment:
`check_canonical()` can't be called off the main thread.

Reply:
https://github.com/AntelopeIO/spring/pull/1708/commits/74ac3486d588cdc62211abbb3eab839b1b5f5c0f